request.go

```diff
@@ -24,7 +24,8 @@ type Request struct {
 
 // NewRequest returns a request initialized with given arguments
 // An non-nil error will be returned if the provided option values are invalid
-func NewRequest(ctx context.Context, path []string, opts OptMap, args []string, file files.Directory, root *Command) (*Request, error) {
+func NewRequest(ctx context.Context, path []string, opts OptMap, args []string,
```

Comment:
I'd like to avoid _mutating_ the passed-in options but this is strictly better than what we had.

Reply:
fixed